Tasks/CargoAuthenticateV0/cargoauthenticatemain.ts

```diff
@@ -0,0 +1,57 @@
+import * as path from 'path';
+import * as tl from 'azure-pipelines-task-lib/task';
+import * as toml from 'toml';
+import * as fs from 'fs';
+import * as constants from './constants';
+
+async function main(): Promise<void> {
+    tl.setResourcePath(path.join(__dirname, 'task.json'));
+
+    try {
+        let configtoml = tl.getInput(constants.CargoAuthenticateTaskInput.WorkingFile);
+        if (!(configtoml.endsWith('.toml'))) {
+            throw new Error(tl.loc('ConfigTomlNotToml', configtoml));
+        }
+        else if (!tl.exist(configtoml)) {
+            throw new Error(tl.loc('ConfigTomlDoesNotExist', configtoml));
+        }
+        else {
+            console.log(tl.loc("AuthenticatingThisConfigToml", configtoml));
+        }
+        
+        // These two format will work
+        // [registries]
+        // zhentan-test = { index = "sparse+https://pkgs.dev.azure.com/codesharing-SU0/zhentan-test/_packaging/zhentan-test/Cargo/index/" }
+        // [registries.zhentan-test1]
+        // index = "sparse+https://pkgs.dev.azure.com/codesharing-SU0/zhentan-test/_packaging/zhentan-test/Cargo/index/"
+        let configtomlFile = fs.readFileSync(configtoml, 'utf8');
+        var result = toml.parse(configtomlFile);
+        if (!result.registries)
+        {
+            throw new Error("config.toml must contains registries field");
+        }
+        var registries = Object.keys(result.registries);
+        var token = `Bearer ${tl.getVariable('System.AccessToken')}`;
+
+        for(let registry of registries) {
```

Comment:
Where do we distinguish which registries are Azure Artifacts registries? I'm not sure it's a good idea to send the System.AccessToken to all registries in the config file.

Reply:
Added a check, basically we check if the url host matches all the url hosts we support similar as npm's implementation.